docs/use-cases/azure.md

```diff
@@ -52,23 +52,129 @@ az aks create \
+(e.g. `ssh.dev.azure.com:v3`).
+Use the [RFC 3986 compatible syntax](https://tools.ietf.org/html/rfc3986#section-3) instead: `ssh.dev.azure.com/v3`.
+
+You will be prompted to add a deploy key to your repository.
+If you don't specify the SSH algorithm, then `flux` will generate an RSA 2048 bits key.
+
+The `flux create source git` command will prompt you to add a deploy key to your repository, but Azure DevOps
+[does not support repository or org-specific deploy keys](https://developercommunity.visualstudio.com/t/allow-the-creation-of-ssh-deploy-keys-for-vsts-hos/365747).
+You may add the deploy key to a user's personal SSH keys being mindful that removing them from the repo may revoke Flux's access.
+As an alternative, create a machine-user whose sole purpose is to store credentials for automation.
+Using a machine-user also has the benefit of being able to be read-only or restricted to specific repositories if that is needed.
+
+If you wish to use Git over HTTPS, then generate a personal access token and supply it as the password:
+
+```sh
+flux create source git flux-system \
+  --git-implementation=libgit2 \
+  --url=https://dev.azure.com/<org>/<project>/_git/<repository> \
+  --branch=master \
+  --username=git \
+  --password=${AZ_PAT_TOKEN} \
+  --interval=1m
+```
+
+Please consult the [Azure DevOps documentation](https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate?view=azure-devops&tabs=preview-page)
+on how to generate personal access tokens for Git repositories.
+Azure DevOps PAT's always have an expiration date, so be sure to have some process for renewing or updating these tokens.
+Similar to the lack of repo-specific deploy keys, a user needs to generate a user-specific PAT.
+If you are using a machine-user, you can generate a PAT or simply use the machine-user's password which does not expire.
+
+Create a `Kustomization` object on your cluster:
+
+```sh
+flux create kustomization flux-system \
+  --source=flux-system \
+  --path="./clusters/my-cluster" \
+  --prune=true \
+  --interval=10m
+```
+
+Export both objects, generate a `kustomization.yaml`, commit and push the manifests to Git:
+
+```sh
+flux export source git flux-system \
+  > ./clusters/my-cluster/flux-system/gotk-sync.yaml
+
+flux export kustomization flux-system \
+  >> ./clusters/my-cluster/flux-system/gotk-sync.yaml
+
+cd ./clusters/my-cluster/flux-system && kustomize create --autodetect
+
+git add -A && git commit -m "add sync manifests" && git push
+```
+
+To upgrade the Flux components to a newer version, download the latest `flux` binary,
+run the install command and commit the changes:
+
+```sh
+flux install \
+  --export > ./clusters/my-cluster/flux-system/gotk-components.yaml
+
+git add -A && git commit -m "update flux" && git push
+```
+
+The source-controller will pull the changes on the cluster, then the kustomize-controller
+will perform a rolling update of all Flux components including itself.
```

Comment:
I think it would probably be better to give this section a heading for people that are scan-reading.